Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -231,30 +242,20 @@ private IBocList GetControlToValidate ()
     void IValidatorWithDynamicErrorMessage.RefreshErrorMessage ()
     {
       var bocList = GetControlToValidate();
-      var validationRepository = bocList.ValidationFailureRepository;
+
+      var context = new ValidationFailureHandlingContext(bocList);
+      _validationFailureHandler.HandleValidationFailures(context);
 
       var errorMessageBuilder = new StringBuilder();
 
-      var validationFailuresForBusinessObjectPropertyOfBocList = validationRepository.GetUnhandledValidationFailuresForBocList(true);
-      foreach (var validationFailure in validationFailuresForBusinessObjectPropertyOfBocList)
-      {
-        errorMessageBuilder.AppendLine(validationFailure.Failure.ErrorMessage);
-      }
+      errorMessageBuilder.Append(context.GetCompoundedErrorMessage());
 
-      var hasRowsWithUnhandledValidationFailures = validationRepository.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true).Any();
-      if (hasRowsWithUnhandledValidationFailures)
+      if (bocList.ValidationFailureRepository.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true).Any())
       {
-        var remainingValidationFailuresText = bocList
-            .GetResourceManager()
-            .GetString(BocList.ResourceIdentifier.ValidationFailuresFoundInOtherListPagesErrorMessage);
-
-        errorMessageBuilder.AppendLine(remainingValidationFailuresText);
+        errorMessageBuilder.Append(bocList.GetResourceManager().GetString(BocList.ResourceIdentifier.ValidationFailuresFoundInListErrorMessage));
       }
 
-      if (errorMessageBuilder.Length > 0)
-        ErrorMessage = errorMessageBuilder.ToString();
-      else
-        ErrorMessage = GetControlToValidate().GetResourceManager().GetString(BocList.ResourceIdentifier.ValidationFailuresFoundInListErrorMessage);
+      ErrorMessage = errorMessageBuilder.ToString();
```

Comment:
I think you have changed the semantic here? Do you already have tests for the Refresh error message method? 

Before, `ValidationFailuresFoundInListErrorMessage` was only appended if there where no other messages in the result.

Reply:
@a-ctor: https://github.com/re-motion/Framework/pull/764#discussion_r1266491136

I think both versions can be correct, I honestly don't know which one we should go for.